qiskit/transpiler/__init__.py

```diff
@@ -155,15 +228,602 @@
+
+The ideal situation for the layout stage is to find a "perfect" layout, where all operations
+respect the connectivity constraints of the :class:`.Target` such that the routing stage
+is not required.  This is typically not possible for arbitrary input circuits, but when it is, the
+:class:`.VF2Layout` pass can be used to find a valid initial layout.  If multiple perfect layouts
+are found, a scoring heuristic based on estimated error rates is used to decide which one to use.
+
+In all built-in plugins, passing the :func:`.generate_preset_pass_manager` argument
+``initial_layout`` causes the given layout to be used verbatim, skipping the individual "choosing"
+logic.  All built-in plugins also handle embedding the circuit into the full width of the device,
+including assigning ancillas.
+
+If you write your own layout plugin, you might find :func:`.generate_embed_passmanager` useful for
+automating the "embedding" stage of the layout application.
+
+When writing :ref:`stage plugins <transpiler-preset-stage-plugins>`, the entry point for ``layout``
+is ``qiskit.transpiler.layout``.  The built-in plugins are:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Method
+      - Summary
+
+    * - :ref:`default <transpiler-preset-stage-layout-default>`
+      - At the highest optimization levels, attempts to find a perfect layout, then tries a
+        Sabre-based layout-and-routing combined pass.
+
+    * - :ref:`dense <transpiler-preset-stage-layout-dense>`
+      - Finds the densest subgraph (in terms of qubit link degrees) of the backend to use as the
+        initial qubits.
+
+    * - :ref:`trivial <transpiler-preset-stage-layout-trivial>`
+      - Maps virtual qubit 0 to physical qubit 0, and so on.
+
+    * - :ref:`sabre <transpiler-preset-stage-layout-sabre>`
+      - Uses `Qiskit's enhanced Sabre layout algorithm <sabre-lightsabre-paper_>`_.
+
+At all optimization levels, the default layout method is ``default``, though the structure of this
+stage changes dramatically based on the level.
+
+.. _transpiler-preset-stage-layout-default:
+
+Built-in ``default`` plugin
+...........................
+
+An amalgamation of several different layout techniques.
+
+At optimization level 0, the trivial layout is chosen.
+
+At optimization levels above 0, there is a two-step process:
+
+#. First, use :class:`.VF2Layout` to attempt to find a "perfect" layout.  The maximum number of
+   calls to the isomorphism evaluator increases with the optimization level.  For huge, complex
+   targets, we are not guaranteed to find perfect layouts even if they exist, but the chance
+   increases with the optimization level.
+
+#. If no perfect layout can be found, use :class:`.SabreLayout` to choose an initial layout, with
+   the numbers of initial layout trials, swap-map trials, and forwards–backwards iterations
+   increasing with the optimization level.
+
+In addition, optimization levels above 0 also try the trivial layout before the VF2-based version,
```

Comment:
This is only for opt level 1, it tries trivial layout first and if that is perfect it skips vf2 and sabre. Level 2 and 3 do not run trivial layout.

Reply:
Huh, I knew that.  I don't know why I wrote it differently.

Reply:
```suggestion
In addition, optimization level 1 also tries the trivial layout before the VF2-based version,
```